### ISSUE TITLE
No matching autoconf version during installation

### ISSUE THREAD

**Issue by VictorForouhar** [rhayes777/PyAutoFit#494]
During installation of the main branch version, pip complains about not finding the requested version of autoconf. 

This is caused by this [line](https://github.com/rhayes777/PyAutoFit/blob/ab0e8bc104b978e7cfc3b3a3542ea5f1a8f7586a/setup.py#L16), which makes pip look for `autoconf==1.0.dev0`. This is solved by changing this line to whatever version of autoconf is available, e.g. for the latest one:
```
requirements.extend([
    f'autoconf==2022.3.18.2'
])
```

**Comment by Jammy2211** [comment]
Were you following these insturctions?

https://pyautofit.readthedocs.io/en/latest/installation/source.html

**Comment by VictorForouhar** [comment]
Unfortunately, no. I directly installed it using 
```
git clone PyAutoFit
cd PyAutoFit
pip install ./
```

I now used the instructions and it has not complained about `autoconf`. Is exporting the path better than installing it as I did because it gives all venv access to PyAutoFit? 

As this time I was following the instructions, I also run `python3 -m pytest`. Although not related to the original issue,  it came back with some errors:
```
============================================================================== short test summary info ===============================================================================
FAILED test_autofit/database/query/test_grid_seach.py::test_grid_search_best_fits - sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) near "WITH": syntax error
FAILED test_autofit/database/query/test_grid_seach.py::test_multiple_best_fits - sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) near "WITH": syntax error
FAILED test_autofit/graphical/stochastic/test_regression.py::test_stochastic_linear_regression - ModuleNotFoundError: jax needed if `factor_vjp` not passed with vjp=True
ERROR test_autofit/graphical/regression/test_linear_regression.py::test_jacobian - ModuleNotFoundError: jax needed if `factor_vjp` not passed with vjp=True
ERROR test_autofit/graphical/regression/test_linear_regression.py::test_laplace_jac - ModuleNotFoundError: jax needed if `factor_vjp` not passed with vjp=True
ERROR test_autofit/graphical/regression/test_logistic_regression.py::test_laplace - ModuleNotFoundError: jax needed if `factor_vjp` not passed with vjp=True
========================================================== 3 failed, 778 passed, 43 warnings, 3 errors in 91.00s (0:01:30) =============================================
```

**Comment by Jammy2211** [comment]
> I now used the instructions and it has not complained about autoconf. Is exporting the path better than installing it as I did because it gives all venv access to PyAutoFit?

I'm not sure, this is just how we've always done it 🤣.

I'll leave this issue open to see if rich has any opinion on how we should set up installs.


Yeah I think we have broken unit tests, these are probably due to a missing optional install:

```
ultranest==3.2.0
jax==0.3.1
jaxlib==0.3.0
```

**Comment by rhayes777** [comment]
So this is happening because we no longer have a development snapshot version on pip. I think the best default behaviour would be to install the latest release from PyPI although there's a risk that if the API of the dependency has changed since the last release that version won't be the version we need.

**Comment by jonathanfrawley** [comment]
Is the problem that there is a dependency on a development branch of a dependent project? My memory was that you would release dependencies (merging them to master and releasing them to Pypi) before depending on them in the release branches of other projects.

**Comment by rhayes777** [comment]
Yeah that's true. I guess in this case it's being installed for development purposes so the solution is probably just to clone all the dependency repo

**Comment by jonathanfrawley** [comment]
Yeah, I think that was the idea since that is what people seem to do when developing anyway.

**Comment by rhayes777** [comment]
Does that make sense for you @VictorForouhar ?

**Comment by VictorForouhar** [comment]
I think so, @rhayes777. For the case of autoconf, do you mean cloning the latest version of this [repo](https://github.com/rhayes777/PyAutoConf) and pointing the requirements for autoconf to the folder where I have cloned it?

**Comment by rhayes777** [comment]
Yes exactly. You would need to clone that repo and add it to your PYTHONPATH
